crates/db_schema/src/source/local_site.rs

```diff
@@ -125,6 +127,12 @@ pub struct LocalSiteInsertForm {
   pub default_post_sort_type: Option<PostSortType>,
   #[new(default)]
   pub default_comment_sort_type: Option<CommentSortType>,
+  #[new(default)]
+  pub oauth_registration: Option<bool>,
+  #[new(default)]
+  pub reject_federated_upvotes: Option<bool>,
+  #[new(default)]
+  pub reject_federated_downvotes: Option<bool>,
```

Comment:
Consider adding a separate struct `VoteOptions` to hold these voting related options (also `enable_downvotes`).

Reply:
That doesn't cover a lot of cases, such as : 

- Allow local votes, but not federated upvotes. 
- Allow local downvotes, but not federated downvotes.

To satisfy #4527, we should just add a `enable_upvotes` boolean alongside the already existing `enable_downvotes` boolean. If  you'd like me to add that in this PR, I can.

Reply:
#4527 is specifically about disabling all votes for comments, but votes on posts should remain enabled.

Reply:
We could add a `disable_comment_voting` boolean, but looking at that issue, its main goal is to decrease DB load. So dissallowing federated votes (like this PR adds) would probably be good enough anyway.

Reply:
I dont think so, without federated post votes on a small instance like slrpnk.net, the active sort and other sorts wouldnt work anymore.

Reply:
> 




On second thought, that could work:

```rust
enum VoteOptions {
  AllowAll,
  AllowLocal
  Disable
}
```

As long as upvotes and downvotes are separate for each also:

```rust
pub struct LocalSiteInsertForm {
  ...,
  post_upvotes: Option<VoteOptions>,
  post_downvotes: Option<VoteOptions>,
  comment_upvotes: Option<VoteOptions>,
  comment_downvotes: Option<VoteOptions>
}
```

Would that be okay with everyone?

Reply:
So basically up and down-votes for posts, respective comments are all completely separate and you can individually set them to be one of the three options?

Might be nice to allow down-votes only locally like that...

Reply:
@dessalines Yes that looks good.

@poVoq With this definition it will be possible to freely configure any possible combination of voting options (post/comment, upvote/downvote, local/federated/none). Only remaining limitation is that this info is not federated, so other instances wont know which voting options are available in a given community. But that can be fixed later in a separate PR.